### PR TITLE
lib: Use Nettle for computing SHA-1 digests

### PR DESCRIPTION
**Pull request by mgrabovsky**
Similarly to abrt/satyr#295.

The `str_to_sha1()` and `str_to_sha1str()` functions from `hash_sha1.c` weren't used anywhere, so I just ripped out the whole part.

### CODE REVIEW

File: src/lib/problem_data.c [hunk 1]
```diff
@@ -143,15 +143,15 @@ void problem_data_add_basics(problem_data_t *pd)
...
             }
             g_list_free(list);
 
             /* end hash */
-            char hash_bytes[SHA1_RESULT_LEN];
-            sha1_end(&sha1ctx, hash_bytes);
-            char hash_str[SHA1_RESULT_LEN*2 + 1];
-            bin2hex(hash_str, hash_bytes, SHA1_RESULT_LEN)[0] = '\0';
+            unsigned char hash_bytes[SHA1_DIGEST_SIZE];
+            sha1_digest(&hash, SHA1_DIGEST_SIZE, hash_bytes);
```
Comment (ernestask) [inline src/lib/problem_data.c]: I’m wondering… Wouldn’t it be better to use `sizeof(hash_bytes)` here and below?

**Comment by ernestask** [comment]
Lovely diff there.

File: src/lib/problem_data.c (hunk 1 above)
Comment (mgrabovsky) [reply src/lib/problem_data.c]: Made your dreams come true.

**Comment by mkutlak** [comment]
Got:
```
abrt-server[108282]: abrt-action-analyze-c: symbol lookup error: abrt-action-analyze-c: undefined symbol: libreport_str_to_sha1str
```

https://github.com/abrt/abrt/search?q=str_to_sha1str

**Comment by ernestask** [comment]
Oops.

**Pull request merged**